Extensions/TileMap/JsExtension.js

```diff
@@ -0,0 +1,662 @@
+          .getValue(),
+        0
+      );
+      const displayMode = this._associatedObject
+        .getProperties(this.project)
+        .get('displayMode')
+        .getValue();
+      const tilesetJsonFile = this._associatedObject
+        .getProperties(this.project)
+        .get('tilesetJsonFile')
+        .getValue();
+
+      PixiTilemapHelper.getPIXITileSet(
+        (textureName) =>
+          this._pixiResourcesLoader.getPIXITexture(this._project, textureName),
+        tilesetJsonData
+          ? { ...tileMapJsonData, tilesets: [tilesetJsonData] }
+          : tileMapJsonData,
+        tilemapAtlasImage,
+        tilemapJsonFile,
+        tilesetJsonFile,
+        (tileset) => {
+          if (tileset && this._pixiObject) {
+            PixiTilemapHelper.updatePIXITileMap(
+              this._pixiObject,
+              tileset,
+              displayMode,
+              layerIndex,
+              pako
+            );
+          }
+        }
+      );
+    };
+
+    /**
+     * This is called to update the TileMap
+     */
+    RenderedTileMapInstance.prototype.updateTileMap = function () {
+      // Get the tileset resource to use
+      const tilemapJsonFile = this._associatedObject
+        .getProperties(this.project)
+        .get('tilemapJsonFile')
+        .getValue();
+      const tilesetJsonFile = this._associatedObject
+        .getProperties(this.project)
+        .get('tilesetJsonFile')
+        .getValue();
+
+      this._pixiResourcesLoader
+        .getResourceJsonData(this._project, tilemapJsonFile)
+        .then((tileMapJsonData) => {
+          if (tilesetJsonFile) {
+            this._pixiResourcesLoader
+              .getResourceJsonData(this._project, tilesetJsonFile)
+              .then((tilesetJsonData) => {
+                this._loadTileMapWithTileset(tileMapJsonData, tilesetJsonData);
+              });
+          } else {
+            this._loadTileMapWithTileset(tileMapJsonData);
+          }
+        });
+    };
+    /**
+     * This is called to update the PIXI object on the scene editor
+     */
+    RenderedTileMapInstance.prototype.update = function () {
+      const tilemapJsonFile = this._associatedObject
+        .getProperties(this.project)
+        .get('tilemapJsonFile')
+        .getValue();
+      if (this._pixiObject.tilemapJsonFile !== tilemapJsonFile)
+        this._pixiObject.tilemapJsonFile = tilemapJsonFile;
+
+      const tilesetJsonFile = this._associatedObject
+        .getProperties(this.project)
+        .get('tilesetJsonFile')
+        .getValue();
+      if (this._pixiObject.tilesetJsonFile !== tilesetJsonFile)
+        this._pixiObject.tilesetJsonFile = tilesetJsonFile;
+
+      const tilemapAtlasImage = this._associatedObject
+        .getProperties(this.project)
+        .get('tilemapAtlasImage')
+        .getValue();
+      if (this._pixiObject.tilemapAtlasImage !== tilemapAtlasImage)
+        this._pixiObject.tilemapAtlasImage = tilemapAtlasImage;
```

Comment:
I'm not sure why you keep insisting putting things on pixiObject - remember it's dangerous to just add stuff on an object that is not supposed to have these :)
What is _pixiObject? It's this object: https://github.com/pixijs/pixi-tilemap/blob/master/src/CompositeRectTileLayer.ts

There is no tilemapAtlasImage, no tilesetJsonFile and no tilemapJsonFile - so do not add them. They seem to be actually... unused in the code ;)

JavaScript is a forgiving language, but this will come back to bite you very very hard (remember we spent multiple days because previously the `render` property was overriden). 
The rule of thumb is simple: if a property is not documented, don't set it.

Reply:
I've removed these in https://github.com/4ian/GDevelop/pull/1901/commits/65769e2be452b11942809bbce4376b18cddea010

See, just removing this code I feel 100x times better ;) No clutter, just a plain, simple, update method that put the object at the right position. Just what it is meant to do! :D